### ISSUE TITLE
[Discussion] how can we establish a feedback loop to learn better from "failures"

### ISSUE THREAD

**Issue by sw-yx** [mixmaxhq/frecency#13]
thanks for this library! i've only just been exploring frecency and read this great blogpost: https://blog.ram.rachum.com/post/33095461928/developers-when-using-frecency-please-consider

and i think its pretty true. recency can be *bad* if too eager. this may involve supplying another key, or adding a different weight, or something I havent thought of. just offering this link, to provoke discussion.

**Comment by wearhere** [comment]
Very interesting, thanks for the link! I don't immediately understand how Gmail would suffer this problem when using frecency, because IIRC (been awhile since I worked on this library 😅) frecency performs exactly like Ram suggests (my emphasis):

> The way to fix this problem is to always check how certain selections are made. 
 If I always start an email to Bob Jones by typing Bob, don’t offer that selection in a high position when I start typing Jones. _Every time I make a selection, save the original piece of text that I typed in_

If Ram selects John after typing "jo", then when next typing "jo", John should sort to the top, not Bob.

I wonder if Gmail is actually using frecency (or, at the least, if this library is vulnerable to the same problem as their implementations are). It sounds like Gmail might be searching by substring (finding "jo" in "Bob Jones") and separately sorting by recency.
